### PR TITLE
Added proxy support using constructor approach

### PR DESCRIPTION
**Pull request by lersouza**
Added proxy support to TMDbClient API.
Using the constructor approach, receiving a IWebProxy instance in case a proxy should be used.

### CODE REVIEW

**Comment by lersouza** [comment]
This pull request Closes #201 .

File: TMDbLib/Client/TMDbClient.cs [hunk 1]
```diff
@@ -108,6 +114,21 @@ public bool ThrowErrorOnExeedingMaxCalls
...
+        /// Gets or sets the Web Proxy to use during requests to TMDb API.
+        /// </summary>
+        /// <remarks>
+        /// The Web Proxy is optional. If set, every request will be sent through it.
+        /// Use the constructor for setting it.
+        /// 
+        /// For convenience, this library also offers a <see cref="IWebProxy"/> implementation.
+        /// Check <see cref="TMDbLib.Utilities.TMDbAPIProxy"/> for more information.
+        /// </remarks>
+        public IWebProxy WebProxy
```
Comment (LordMike) [inline TMDbLib/Client/TMDbClient.cs]: Use an automatic property here. 
public IWebProxy WebProxy { get; }

or:
public IWebProxy WebProxy { get; private set; }

File: TMDbLib/Rest/RestClient.cs [hunk 2]
```diff
@@ -2,25 +2,29 @@
...
             DefaultQueryString = new List<KeyValuePair<string, string>>();
 
             MaxRetryCount = 0;
+            _proxy = proxy;
         }
 
         internal Uri BaseUrl { get; }
         internal List<KeyValuePair<string, string>> DefaultQueryString { get; }
         internal Encoding Encoding { get; } = new UTF8Encoding(false);
+        internal IWebProxy Proxy { get { return _proxy; } }
```
Comment (LordMike) [inline TMDbLib/Rest/RestClient.cs]: Also, an automatic property

File: TMDbLib/Utilities/TMDbAPIProxy.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+namespace TMDbLib.Utilities
+{
+    /// <summary>
+    /// Represents a Web Proxy to use for TMDb API Requests.
+    /// </summary>
+    /// <remarks>
+    /// This is a very simple implementation of a Web Proxy to be used when requesting data from TMDb API.
+    /// It does not support proxy bypassing or multi-proxy configuration based on the destination URL, for instance.
+    /// </remarks>
+    public class TMDbAPIProxy : IWebProxy
```
Comment (LordMike) [inline TMDbLib/Utilities/TMDbAPIProxy.cs]: Isn't there a built in proxy class much like this?

**Comment by LordMike** [comment]
It seems like the proxy implementation you made can be skipped. .NET already has one (https://msdn.microsoft.com/en-us/library/system.net.webproxy(v=vs.110).aspx). 

In any case, if a user wants to use a proxy, they'll figure out how to live up to the `IWebProxy` interface.

File: TMDbLib/Utilities/TMDbAPIProxy.cs (hunk 3 above)
Comment (lersouza) [inline TMDbLib/Utilities/TMDbAPIProxy.cs]: For .NET Core, there isn't. So I decided to build this class.

**Comment by lersouza** [comment]
This approach (https://msdn.microsoft.com/en-us/library/system.net.webproxy(v=vs.110).aspx) is not compatible with .NET Core.

**Pull request merged**